x/tokenomics/keeper/token_logic_modules.go

```diff
@@ -0,0 +1,443 @@
+	logger.Info(fmt.Sprintf("minted (%v) coins in the supplier module", settlementCoins))
+
+	// Send the newley minted uPOKT from the supplier module account
+	// to the supplier's account.
+	if err = k.bankKeeper.SendCoinsFromModuleToAccount(
+		ctx, suppliertypes.ModuleName, supplierAddr, sdk.NewCoins(settlementCoins),
+	); err != nil {
+		return tokenomicstypes.ErrTokenomicsSupplierModuleSendFailed.Wrapf(
+			"sending (%s) to supplier with address %s: %v",
+			settlementCoins,
+			supplier.Address,
+			err,
+		)
+	}
+	logger.Info(fmt.Sprintf("sent (%v) from the supplier module to the supplier account with address %q", settlementCoins, supplier.Address))
+
+	// TODO_MAINNET: Decide on the behaviour here when an app is over serviced.
+	// If an app has 10 POKT staked, but the supplier earned 20 POKT. We still
+	// end up minting 20 POKT but only burn 10 POKT from the app. There are
+	// questions and nuance here that needs to be addressed.
+
+	// Verify that the application has enough uPOKT to pay for the services it consumed
+	if application.GetStake().IsLT(settlementCoins) {
+		settlementCoins, err = k.handleOverservicedApplication(ctx, application, settlementCoins)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Burn uPOKT from the application module account which was held in escrow
+	// on behalf of the application account.
+	if err = k.bankKeeper.BurnCoins(
+		ctx, apptypes.ModuleName, sdk.NewCoins(settlementCoins),
+	); err != nil {
+		return tokenomicstypes.ErrTokenomicsApplicationModuleBurn.Wrapf("burning %s from the application module account: %v", settlementCoins, err)
+	}
+	logger.Info(fmt.Sprintf("burned (%v) from the application module account", settlementCoins))
+
+	// Update the application's on-chain stake
+	newAppStake, err := application.Stake.SafeSub(settlementCoins)
+	if err != nil {
+		return tokenomicstypes.ErrTokenomicsApplicationNewStakeInvalid.Wrapf("application %q stake cannot be reduced to a negative amount %v", application.Address, newAppStake)
+	}
+	application.Stake = &newAppStake
+	logger.Info(fmt.Sprintf("updated application %q stake to %v", application.Address, newAppStake))
+
+	return nil
+}
+
+// TokenLogicModuleGlobalMint processes the business logic for the GlobalMint TLM.
+// TODO_UPNEXT(@olshansk): Delete this in favor of a real TLM that mints tokens
+// and distributes them to the appropriate accounts via boosts.
+func (k Keeper) TokenLogicModuleGlobalMint(
+	ctx context.Context,
+	service *sharedtypes.Service,
+	application *apptypes.Application,
+	supplier *sharedtypes.Supplier,
+	settlementCoins cosmostypes.Coin,
+	relayMiningDifficulty *tokenomictypes.RelayMiningDifficulty,
+) error {
+	logger := k.Logger().With("method", "TokenLogicModuleGlobalMint")
+
+	// Determine how much new uPOKT to mint based on global inflation
+	settlementAmtFloat := new(big.Float).SetUint64(settlementCoins.Amount.Uint64())
+	newMintAmtFloat := settlementAmtFloat.Mul(settlementAmtFloat, big.NewFloat(MintGlobalAllocation))
```

Comment:
Consider a `new(big.Float)` to avoid mutating `settlementAmtFloat`
```suggestion
	newMintAmtFloat := new (big.Float).Mul(settlementAmtFloat, big.NewFloat(MintGlobalAllocation))
```